spec/defines/source_spec.rb

```diff
@@ -387,5 +387,41 @@
       end
     end
 
+    context "with notify_update = true (default)" do
```

Comment:
This is not true. You're using `undef` as the default which, by virtue of `apt::setting` have it set to `true` by default means it turns out to be true. Your test however isn't testing the default chain, it's explicitly setting it as `true`.